internal/resolution/resolver_test.go

```diff
@@ -19,112 +18,114 @@ import (
-		resources := []client.Object{
-			&v1alpha1.Operator{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "prometheus",
-				},
-				Spec: v1alpha1.OperatorSpec{
-					PackageName: "prometheus",
-				},
-			},
-			&v1alpha1.Operator{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "packageA",
-				},
-				Spec: v1alpha1.OperatorSpec{
-					PackageName: "packageA",
-				},
-			},
-		}
-		client := FakeClient(resources...)
-		entitySource := input.NewCacheQuerier(testEntityCache)
-		resolver := resolution.NewOperatorResolver(client, entitySource)
-		solution, err := resolver.Resolve(context.Background())
-		Expect(err).ToNot(HaveOccurred())
-		// 2 * required package variables + 2 * bundle variables
-		Expect(solution.SelectedVariables()).To(HaveLen(4))
-
-		Expect(solution.IsSelected("operatorhub/packageA/2.0.0")).To(BeTrue())
-		Expect(solution.IsSelected("operatorhub/prometheus/0.47.0")).To(BeTrue())
-		Expect(solution.IsSelected("required package packageA")).To(BeTrue())
-		Expect(solution.IsSelected("required package prometheus")).To(BeTrue())
-
-		Expect(solution.IsSelected("operatorhub/prometheus/0.37.0")).To(BeFalse())
-
-	})
-
-	It("should not return an error if there are no Operator resources", func() {
-		var resources []client.Object
-		client := FakeClient(resources...)
-		entitySource := input.NewCacheQuerier(testEntityCache)
-		resolver := resolution.NewOperatorResolver(client, entitySource)
-		solution, err := resolver.Resolve(context.Background())
-		Expect(err).ToNot(HaveOccurred())
-		Expect(solution.SelectedVariables()).To(HaveLen(0))
-	})
-
-	It("should return an error if the entity source throws an error", func() {
-		resource := &v1alpha1.Operator{
+	testResource := []client.Object{
+		&v1alpha1.Operator{
 			ObjectMeta: metav1.ObjectMeta{
 				Name: "prometheus",
 			},
 			Spec: v1alpha1.OperatorSpec{
 				PackageName: "prometheus",
 			},
-		}
-		client := FakeClient(resource)
-		entitySource := FailEntitySource{}
-		resolver := resolution.NewOperatorResolver(client, entitySource)
-		solution, err := resolver.Resolve(context.Background())
-		Expect(solution).To(BeNil())
-		Expect(err).To(HaveOccurred())
-	})
-
-	It("should return an error if the client throws an error", func() {
-		client := NewFailClientWithError(fmt.Errorf("something bad happened"))
-		entitySource := input.NewCacheQuerier(testEntityCache)
-		resolver := resolution.NewOperatorResolver(client, entitySource)
-		solution, err := resolver.Resolve(context.Background())
-		Expect(solution).To(BeNil())
-		Expect(err).To(HaveOccurred())
-	})
-})
+		},
+		&v1alpha1.Operator{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "packageA",
+			},
+			Spec: v1alpha1.OperatorSpec{
+				PackageName: "packageA",
+			},
+		},
+	}
+
+	for _, tt := range []struct {
+		Name              string
+		Client            client.Client
+		EntitySource      input.EntitySource
+		SelectedVariables []deppy.Identifier
+		ExpectedError     error
```

Comment:
Also minor things (no need to change):

* These fields in don't have to be exported as they are in the anonimous struct & never exported
* To distinguish between inputs and expected results I would prefix all the expected data with `want` or `expected`. E.g.:

```suggestion
		name                      string
		client                    client.Client
		entitySource              input.EntitySource
		expectedselectedVariables []deppy.Identifier
		expectedError             error
```